iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/balancer/RouteBalancer.java

```diff
@@ -184,6 +187,24 @@ private Map<TConsensusGroupId, Pair<Integer, Integer>> balanceRegionLeader(
     if (requestId.get() > 0) {
       // Don't retry ChangeLeader request
       AsyncDataNodeClientPool.getInstance().sendAsyncRequestToDataNode(clientHandler);
+      for (int i = 0; i < requestId.get(); i++) {
+        if (clientHandler.getResponseMap().get(i).getCode()
+            == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+          getLoadManager()
+              .forceUpdateRegionLeader(
+                  clientHandler.getRequest(i).getRegionId(),
+                  clientHandler.getRequest(i).getNewLeaderNode().getDataNodeId());
+          LOGGER.info(
+              "[forceUpdateRegionLeader], regionId:{}, datanodeId:{}",
+              clientHandler.getRequest(i).getRegionId(),
+              clientHandler.getRequest(i).getNewLeaderNode().getDataNodeId());
```

Comment:
Remove these logs after u broadcast the RouteChangeEvent. The observer will log those successful leader change event.